### PR TITLE
[test_nn] split convolution tests from test_nn

### PR DESCRIPTION
**Pull request by kshitij12345**
Ref #63085

### CODE REVIEW

File: test/nn/test_convolution.py [hunk 1]
```diff
@@ -0,0 +1,2480 @@
...
+            y_cuda = conv_cuda(x_cpu.to(device))
+            y_cuda.backward(y.to(device))
+
+        self.assertEqual(y_cpu, y_cuda, atol=1e-5, rtol=0, exact_device=False)
+        self.assertEqual(conv_cpu.bias.grad.data, conv_cuda.bias.grad.data, atol=1e-5, rtol=0, exact_device=False)
+        self.assertEqual(conv_cpu.weight.grad.data, conv_cuda.weight.grad.data, atol=1e-5, rtol=0, exact_device=False)
+
+    @onlyCUDA
+    def test_ConvTranspose3d_size_1_kernel(self, device):
+        with set_default_dtype(torch.double):
```
Comment (kshitij12345) [inline test/nn/test_convolution.py:2035]: Default dtype set to double

File: test/nn/test_convolution.py [hunk 2]
```diff
@@ -0,0 +1,2480 @@
...
+        self.assertEqual(m.weight.grad.data,
+                         torch.cat([m1.weight.grad.data, m2.weight.grad.data], 0),
+                         atol=dtype2prec_DONTUSE[torch.float],
+                         rtol=dtype2prec_DONTUSE[torch.float])
+        self.assertEqual(m.bias.grad.data,
+                         torch.cat([m1.bias.grad.data, m2.bias.grad.data], 0),
+                         atol=dtype2prec_DONTUSE[torch.float], rtol=dtype2prec_DONTUSE[torch.float])
+
+    def test_conv_tbc(self):
+        with set_default_dtype(torch.double):
```
Comment (kshitij12345) [inline test/nn/test_convolution.py:622]: with set_default_dtype(torch.double):

**Review by albanD — APPROVED** [review]
Sounds good!

**Comment by kshitij12345** [comment]
Periodic Slow Grad Check is failing (https://github.com/pytorch/pytorch/actions/runs/3326254738/jobs/5503498643#logs) Will investigate.

**Comment by kshitij12345** [comment]
Periodic Slow Gradcheck was already close to the timeout and was disabled: https://github.com/pytorch/pytorch/pull/88008

**Pull request closed without merging**